twitter_stream_block.py

```diff
@@ -81,15 +82,14 @@ class TwitterStreamBlock(Block):
 
     def __init__(self):
         super().__init__()
-        self._result_signals = []
-        self._diagnostic_signals = []
-        self._limit_signals = []
-        self._result_lock = Lock()
-        self._diagnostic_lock = Lock()
-        self._limit_lock = Lock()
+        self._result_signals = defaultdict(list)
+        self._result_lock = defaultdict(Lock)
+        self._lock_lock = Lock()
         self._stop_event = Event()
         self._stream = None
         self._last_rcv = datetime.utcnow()
+        self._limit_count = 0
+        self._limit_lock = Lock()
```

Comment:
Yeah, but I do have something else I was thinking about. Limit notifications can come pretty often from twitter. What happens when two limit notifications try to update _limit_count at the same time? Maybe something to add as an issue for the block?